streams/src/test/java/org/apache/kafka/streams/integration/KStreamRepartitionIntegrationTest.java

```diff
@@ -0,0 +1,859 @@
+               .repartition(repartitioned)
+               .to(outputTopic);
+
+        startStreams(builder);
+
+        final String topic = toRepartitionTopicName(repartitionName);
+
+        validateReceivedMessages(
+            new IntegerDeserializer(),
+            new StringDeserializer(),
+            expectedRecords
+        );
+
+        assertTrue(topicExists(topic));
+        assertEquals(expectedRecords.size(), partitionerInvocation.get());
+    }
+
+    @Test
+    public void shouldPerformSelectKeyWithRepartitionOperation() throws ExecutionException, InterruptedException {
+        final long timestamp = System.currentTimeMillis();
+
+        sendEvents(
+            timestamp,
+            Arrays.asList(
+                new KeyValue<>(1, "10"),
+                new KeyValue<>(2, "20")
+            )
+        );
+
+        final StreamsBuilder builder = new StreamsBuilder();
+
+        builder.stream(inputTopic, Consumed.with(Serdes.Integer(), Serdes.String()))
+               .selectKey((key, value) -> Integer.valueOf(value))
+               .repartition()
+               .to(outputTopic);
+
+        startStreams(builder);
+
+        validateReceivedMessages(
+            new IntegerDeserializer(),
+            new StringDeserializer(),
+            Arrays.asList(
+                new KeyValue<>(10, "10"),
+                new KeyValue<>(20, "20")
+            )
+        );
+
+        final String topology = builder.build().describe().toString();
+
+        assertEquals(1, countOccurrencesInTopology(topology, "Sink: .*-repartition.*"));
+    }
+
+    @Test
+    public void shouldCreateRepartitionTopicIfKeyChangingOperationWasNotPerformed() throws ExecutionException, InterruptedException {
+        final String repartitionName = "dummy";
+        final long timestamp = System.currentTimeMillis();
+
+        sendEvents(
+            timestamp,
+            Arrays.asList(
+                new KeyValue<>(1, "A"),
+                new KeyValue<>(2, "B")
+            )
+        );
+
+        final StreamsBuilder builder = new StreamsBuilder();
+
+        builder.stream(inputTopic, Consumed.with(Serdes.Integer(), Serdes.String()))
+               .repartition(Repartitioned.as(repartitionName))
+               .to(outputTopic);
+
+        startStreams(builder);
+
+        validateReceivedMessages(
+            new IntegerDeserializer(),
+            new StringDeserializer(),
+            Arrays.asList(
+                new KeyValue<>(1, "A"),
+                new KeyValue<>(2, "B")
+            )
+        );
+
+        final String topology = builder.build().describe().toString();
+
+        assertTrue(topicExists(toRepartitionTopicName(repartitionName)));
+        assertEquals(1, countOccurrencesInTopology(topology, "Sink: .*dummy-repartition.*"));
+    }
+
+    @Test
+    public void shouldPerformKeySelectOperationWhenRepartitionOperationIsUsedWithKeySelector() throws ExecutionException, InterruptedException {
```

Comment:
Not sure why we need this test?

Reply:
Wanted to verify that key changing operation with `repartition` works as expected. I think it adds value, especially considering the fact that we've removed `repartition(KeySelector` overloads.

Reply:
Ok. Thanks for clarifying.